### ISSUE TITLE
Remove stream included list check in destroy_stream [BUG]

### ISSUE THREAD

**Issue by zhimingxie** [grpc/grpc#10516]
In `destroy_stream_locked`
`if (s->included[i]) {
      gpr_log(GPR_ERROR, "%s stream %d still included in list %d",
              t->is_client ? "client" : "server", s->id, i);
      abort();
    }`

The abort() can really happen although I didn't find more detail info on it now.
Also I think this abort is meaningless. It doesn't imply other stream or transport is damaged.

So why can we do something like the `grpc_chttp2_list_remove_stalled_by_stream(t, s);` `grpc_chttp2_list_remove_stalled_by_transport(t, s);` above to remove other stream list id ??

**Comment by zhimingxie** [comment]
I can find a code path to reproduce this abort now.

Suppose the following scenario.

In server side, after prepare the response , start to send initialmetadata/message.  
It will schedule` write_action_begin_locked` and `grpc_chttp2_begin_write` to change stream from writable to writing state.

At the time, the client send the rst stream to cancel the call then server  will `read_action_locked` to parse  the new coming data and handle it by `grpc_chttp2_mark_stream_closed` terminally. 
If  `grpc_chttp2_mark_stream_closed` happens before ` write_action_end_locked`, the stream state won't change back to writing  again.  So you will meet the abort in the `destroy_stream_locked`.

In summary, I think it is  a bug   and we need pop/remove the stream from the transport list in `grpc_chttp2_mark_stream_closed`  or  `destroy_stream_locked` or somewhere else.

Please correct me if something incorrect.

**Comment by ncteisen** [comment]
Could you provide a code snippet or a test that demonstrated this behavior? We have hit this abort before, and all times it has caught real bugs that needed to be fixed, so I am hesitant to remove this without a test that uses gRPC correctly and also triggers this abort

**Comment by ncteisen** [comment]
Closing for now. Please reopen if you have a reproduction in which **valid** gRPC code triggers this abort
